rclcpp/include/rclcpp/node_interfaces/node_base.hpp

```diff
@@ -104,9 +104,13 @@ class NodeBase : public NodeBaseInterface
   get_associated_with_executor_atomic() override;
 
   RCLCPP_PUBLIC
-  rcl_guard_condition_t *
+  const rcl_guard_condition_t *
```

Comment:
Why is this needed? This technically breaks API.

Reply:
I removed this change in dd4ab15. For that I had to add a non-const version of `get_rcl_guard_condition` API from `GuardCondition` class